pkg/pipeline/manager.go

```diff
@@ -53,16 +53,20 @@ type Manager struct {
 	logger   logrus.FieldLogger
 }
 
-func (m *Manager) StartPipeline() {
+func (m *Manager) StartPipeline() error {
 	m.logger.Info("starting pipeline... ")
+
+	if len(m.pipeline) == 0 {
```

Comment:
I'm wondering if we could add the check already to the `NewManager` funciton, this way it would be checked even when using the `--check-config` flag to validate configuration. On the other hand maybe it's really just runtime issue and the configuration can be still validated :thinking:

Reply:
I was initially about to inject test into `NewManager()` but changed my mind as soon as found  `manager.addModuleToPipelineEnd(...)` method which makes following pattern legal (used in tests):

```go
manager, err := NewManager(...)
...
err = manager.addModuleToPipelineEnd(...)
...
```

This pattern forces me to add check in `manager.StartPipeline()` allowing `NewManager` pass even without pipelines specified. Does it make sense to you?

Reply:
I can see your point `./slo_exporter --config-file /tmp/empty-file.yaml --check-config` still claims even empty configuration is valid, will address that in additional PR.